eth/db/account.py

```diff
@@ -75,142 +73,11 @@
-        storage root (and therefore will not persist deletes to the base database).
-        The account's storage root must be explicitly set to the empty root.
-        """
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    #
-    # Nonce
-    #
-    @abstractmethod
-    def get_nonce(self, address: Address) -> int:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    @abstractmethod
-    def set_nonce(self, address: Address, nonce: int) -> None:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    #
-    # Balance
-    #
-    @abstractmethod
-    def get_balance(self, address: Address) -> int:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    @abstractmethod
-    def set_balance(self, address: Address, balance: int) -> None:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    #
-    # Code
-    #
-    @abstractmethod
-    def set_code(self, address: Address, code: bytes) -> None:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    @abstractmethod
-    def get_code(self, address: Address) -> bytes:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    @abstractmethod
-    def get_code_hash(self, address: Address) -> Hash32:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    @abstractmethod
-    def delete_code(self, address: Address) -> None:
-        raise NotImplementedError("Must be implemented by subclasses")
-
-    #
-    # Account Methods
-    #
-    @abstractmethod
-    def account_is_empty(self, address: Address) -> bool:
-        raise NotImplementedError("Must be implemented by subclass")
-
-    #
-    # Record and discard API
-    #
-    @abstractmethod
-    def record(self) -> JournalDBCheckpoint:
-        raise NotImplementedError("Must be implemented by subclass")
-
-    @abstractmethod
-    def discard(self, checkpoint: JournalDBCheckpoint) -> None:
-        raise NotImplementedError("Must be implemented by subclass")
-
-    @abstractmethod
-    def commit(self, checkpoint: JournalDBCheckpoint) -> None:
-        raise NotImplementedError("Must be implemented by subclass")
-
-    @abstractmethod
-    def make_state_root(self) -> Hash32:
-        """
-        Generate the state root with all the current changes in AccountDB
-
-        Current changes include every pending change to storage, as well as all account changes.
-        After generating all the required tries, the final account state root is returned.
-
-        This is an expensive operation, so should be called as little as possible. For example,
-        pre-Byzantium, this is called after every transaction, because we need the state root
-        in each receipt. Byzantium+, we only need state roots at the end of the block,
-        so we *only* call it right before persistance.
-
-        :return: the new state root
```

Comment:
Anywhere like this where the original abc method had a docstring, the docstring was migrated to the new abc base class.